lib/dispatch-rider/logging/json_formatter.rb

```diff
@@ -0,0 +1,74 @@
+require 'json'
+
+module DispatchRider
+  module Logging
+    class JsonFormatter < BaseFormatter
+      def format_error_handler_fail(message, exception)
+        as_json do
+          {
+            message: "Failed error handling",
+          }.merge exception_info_fragment(message, exception)
+        end
+      end
+
+      def format_got_stop(message, reason)
+        as_json do
+          {
+            message: "Got stop",
+            reason: reason,
+          }.merge message_info_fragment(message)
+        end
+      end
+
+      def format_handling(kind, message, exception: nil, duration: nil)
+        as_json do
+          case kind
+          when :start
+            { message: "Starting execution" }.merge message_info_fragment(message)
```

Comment:
maybe instead of message use a more machine parseable name .. eg: `phase: 'start'`